frontend/src/components/UpgradeDialog.tsx

```diff
@@ -0,0 +1,122 @@
+import { LifeBuoy, Mail, RefreshCw, Users, Zap } from "lucide-react";
+import { ReactNode } from "react";
+import { Button } from "src/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "src/components/ui/dialog";
+import { ALBY_PRO_PLAN } from "src/constants";
+import { useAlbyMe } from "src/hooks/useAlbyMe";
+import { openLink } from "src/utils/openLink";
+
+interface UpgradeDialogProps {
+  children: ReactNode;
+}
+
+export const UpgradeDialog = ({ children }: UpgradeDialogProps) => {
+  const { data: albyMe } = useAlbyMe();
+
+  const handleUpgrade = () => {
+    openLink(
+      `https://www.getalby.com/subscription/confirm?plan=${ALBY_PRO_PLAN}`
+    );
+  };
+
+  if (albyMe?.subscription.buzz) {
```

Comment:
to check if we should use `buzz` or some other property instead like if the plan code exists

Reply:
@reneaaron I can have a look at this tomorrow

Reply:
I think it's good enough for now.